src/main/java/de/numcodex/feasibility_gui_backend/api/QueryHandlerRestController.java

```diff
@@ -54,6 +58,16 @@ public Response runQuery(
             .pathSegment("api", "v1", "query-handler", "result", id)
             .build()
             .toUri();
+
+    if(apiBaseUrl.length() > 0){
+      uri =
+          ServletUriComponentsBuilder.fromUriString(apiBaseUrl)
+              .replacePath("")
+              .pathSegment("api", "v1", "query-handler", "result", id)
+              .build()
+              .toUri();
+    }
+
```

Comment:
Reassigning another value to `uri` is unnecessary.
Instead we can shorten this to something like:

```suggestion
UriComponentsBuilder uriBuilder = (apiBaseUrl != null && !apiBaseUrl.isEmpty())
            ? ServletUriComponentsBuilder.fromUriString(apiBaseUrl)
            : ServletUriComponentsBuilder.fromRequestUri(httpServletRequest);

    var uri = uriBuilder.replacePath("")
            .pathSegment("api", "v1", "query-handler", "result", id)
            .build()
            .toUri();
```